src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala

```diff
@@ -207,6 +207,19 @@ final case class AgreementsApiServiceImpl(
     }
   }
 
+  override def archiveAgreement(
+    agreementId: String
+  )(implicit contexts: Seq[(String, String)], toEntityMarshallerProblem: ToEntityMarshaller[Problem]): Route = {
+    val result: Future[Unit] = for {
+      agreementUuid <- agreementId.toFutureUUID
+      _             <- agreementProcessService.archiveAgreement(agreementUuid)
+    } yield ()
```

Comment:
You're an alumni of @beetlecrunch school of for-comprehensions!

Reply:
The problem is the following: Cats cannot be used anymore
//Not passed
val result: Future[Unit] = agreementId.toFutureUUID.flatMap(agreementProcessService.archiveAgreement).void 
//Ugly
val result: Future[Unit] = agreementId.toFutureUUID.flatMap(agreementProcessService.archiveAgreement).map(_ => ()) 
//Nope
val result: Future[Unit] = (agreementId.toFutureUUID >>= agreementProcessService.archiveAgreement).void
//Clear and idiomatic style
val result: Future[Unit] = for {
      agreementUuid <- agreementId.toFutureUUID
      _             <- agreementProcessService.archiveAgreement(agreementUuid)
    } yield ()